src/python/pants/backend/python/goals/coverage_py.py

```diff
@@ -286,33 +286,42 @@ class InvalidCoverageConfigError(Exception):
     pass
 
 
-def _update_config(fc: FileContent) -> FileContent:
-    if PurePath(fc.path).suffix == ".toml":
-        try:
-            all_config = toml.loads(fc.content.decode())
-        except toml.TomlDecodeError as exc:
-            raise InvalidCoverageConfigError(
-                f"Failed to parse the coverage.py config `{fc.path}` as TOML. Please either fix "
-                f"the config or update `[coverage-py].config` and/or "
-                f"`[coverage-py].config_discovery`.\n\nParse error: {repr(exc)}"
-            )
-        tool = all_config.setdefault("tool", {})
-        coverage = tool.setdefault("coverage", {})
-        run = coverage.setdefault("run", {})
-        run["relative_files"] = True
-        if "pytest.pex/*" not in run.get("omit", []):
-            run["omit"] = [*run.get("omit", []), "pytest.pex/*"]
-        return FileContent(fc.path, toml.dumps(all_config).encode())
+def _parse_toml_config(fc: FileContent) -> MutableMapping[str, Any]:
+    try:
+        return toml.loads(fc.content.decode())
+    except toml.TomlDecodeError as exc:
+        raise InvalidCoverageConfigError(
+            f"Failed to parse the coverage.py config `{fc.path}` as TOML. Please either fix "
+            f"the config or update `[coverage-py].config` and/or "
+            f"`[coverage-py].config_discovery`.\n\nParse error: {repr(exc)}"
+        )
 
+
+def _parse_ini_config(fc: FileContent) -> configparser.ConfigParser:
     cp = configparser.ConfigParser()
     try:
         cp.read_string(fc.content.decode())
+        return cp
     except configparser.Error as exc:
         raise InvalidCoverageConfigError(
             f"Failed to parse the coverage.py config `{fc.path}` as INI. Please either fix "
             f"the config or update `[coverage-py].config` and/or `[coverage-py].config_discovery`."
             f"\n\nParse error: {repr(exc)}"
         )
+
+
+def _update_config(fc: FileContent) -> FileContent:
```

Comment:
This is unchanged in functionality, it just delegates the parsing to the two new helper functions.